src/app/api/staking/getStakedDeposits.ts

```diff
@@ -29,8 +30,9 @@ export async function fetchStakedDepositsForAddress({
 
   return deposits.map((deposit) => {
     return {
-      //Note: Large amounts are stored in scientific notation 1.002e+21
-      amount: Number(deposit.amount).toLocaleString().replace(/,/g, ""),
+      amount: scientificNotationToPrecision(
+        deposit.amount.toString()
```

Comment:
@andreitr wondering if you tried using `deposit.amount.toFixed(0)`? This should technically convert prisma numeric to a properly formatted number string with 0 decimal places

Reply:
I ended up writing a custom scientificNotationToPrecision method which seems to be better at handling edge cases.